src/Modal/Portal.js

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+import styles from "./Modal.module.scss";
+
+class Portal extends React.Component {
+  constructor() {
+    super();
+
+    this.portal = React.createRef();
+  }
+
+  componentDidMount() {
+    this.portal.current.focus();
+  }
+
+  handleOnKeyUp = e => {
+    if (e.keyCode == 27) {
+      this.props.closeModal();
+    }
+  };
+
+  handleOnClick = e => {
+    if (this.portal.current == e.target) {
+      this.props.closeModal();
+    }
+  };
+
+  render() {
+    const { children } = this.props;
+
+    return ReactDOM.createPortal(
+      <div
+        tabIndex={0}
+        ref={this.portal}
+        className={styles.container}
+        onClick={this.handleOnClick}
+        onKeyUp={this.handleOnKeyUp}
+      >
+        <div className={styles.dialog}>
+          {children}
+        </div>
+      </div>,
+      document.body
```

Comment:
Mounting directly into the `body` is a bad habit. Much better is to create independent `div`. The example can be found [here](https://github.com/santiment/app/blob/3f32bd546d7a4500ee01b3243d6e9e6cc7bd92e4/src/components/MultilineText/MultilineText.js#L4-L14)

Reply:
Using `id` problems:
1. There is a possibility, that library user can by chance use the same `id` in his code base. Bad from the HTML standard point of view;
2. Still having checks which are dependent on some arbitrary value.

On the other side, `div.dimmed` could have such props `onClick={this.closeModal`} and `onKeyPress={this.onEscKeyPress}`

Reply:
@DmitriVanGuard the issue onClick event is that is will propagate through the dialog box as well. So even when I click the dialog box, the onClick will be in action.

Reply:
In the proposed markup they are sibling 
```html
<div#ui-modal>
  <div.dialog> 
  <div.dimmed> // dimmed area with onClick and onKeyPress handle for closing
</div>
```